libcxx/test/support/count_new.h

```diff
@@ -99,9 +99,10 @@ class MemCounter
 
     void deleteCalled(void * p)
     {
-        assert(p);
-        --outstanding_new;
-        ++delete_called;
+        if (p) {
+          --outstanding_new;
+          ++delete_called;
+        }
     }
 
     void alignedDeleteCalled(void *p, std::size_t a) {
```

Comment:
This one needs to be fixed too, since it's valid to call the aligned delete with a nullptr. So this should be

```c++
if (p) {
  deleteCalled(p);
  ++aligned_delete_called;
  last_delete_align = a;
}
```